examples/using-gatsby-image/src/components/navigation.js

```diff
@@ -1,84 +1,199 @@
 import React from "react"
 import { Link } from "gatsby"
+import styled, { css } from "react-emotion"
+import { MdLaunch } from "react-icons/md"
 
-import { rhythm, scale, options } from "../utils/typography"
+import { rhythm, options } from "../utils/typography"
+import presets, { colors } from "../utils/presets"
+
+const linkFontSize = {
+  [presets.Phablet]: {
+    fontSize: 20,
+  },
+  [presets.Tablet]: {
+    fontSize: 24,
+  },
+  [presets.Desktop]: {
+    fontSize: 28,
+  },
+}
+
+const linkStyle = {
+  fontFamily: options.headerFontFamily.join(`,`),
+  letterSpacing: `.005em`,
+  transition: `all 200ms cubic-bezier(.17, .67, .83, .67)`,
+  whiteSpace: `nowrap`,
+  fontWeight: 700,
+  position: `relative`,
+  "@media (-webkit-min-device-pixel-ratio:0)": {
+    backgroundImage: `linear-gradient(45deg, #EB4D9C, #D33024, #E48233, #F4E24D, #B4DC48, #54B2EA, ${
+      colors.gatsby
+    } 50%)`,
+    backgroundPosition: `100% 0`,
+    backgroundSize: `200% 200%`,
+    color: `rgba(102, 51, 153, 1)`,
+    WebkitBackgroundClip: `text`,
+    "&:hover": {
+      backgroundPosition: `0 0`,
+      color: `rgba(102, 51, 153, 0) !important`,
+      WebkitTextFillColor: `transparent`,
+    },
+  },
+}
+
+const assignActiveStyles = ({ isPartiallyCurrent }) =>
+  isPartiallyCurrent
+    ? {
+        className: css({
+          ...linkStyle,
+          color: `rgba(0, 0, 0, 1) !important`,
+          "&&": {
+            "&:before": {
+              content: `""`,
+              position: `absolute`,
+              bottom: 0,
+              left: 0,
+              right: 0,
+              width: 12,
+              height: 2,
+              backgroundColor: `#000`,
+            },
+            [presets.Desktop]: {
+              "&:before": {
+                width: 18,
+                height: 4,
+              },
+            },
+          },
+
+          "&:hover": {
+            backgroundPosition: `100% 0 !important`,
+            color: `initial !important`,
+            WebkitTextFillColor: `initial !important`,
+          },
+        }),
+      }
+    : { className: css({ "&&": { ...linkStyle } }) }
+
+const Header = styled(`header`)({
```

Comment:
another nit, but I'd probably always prefer the shorthand, e.g. `styled.header` vs. the function call.

More of a personal preference here, but I think it's _probably_ worth following that.

Reply:
@jlengstorf this is why we need a style definition i think. pretty sure elsewhere people have said the ('...') style is preferred.

Reply:
Yeah, @DSchau we talked about this in another thread and had started using `styled('header')` as a standard. We can change that if there's a strong opinion either way — we just want to keep it consistent everywhere.

Reply:
@amberleyromo  @DSchau  @jlengstorf  @fk 

I'm not sure if there is a way to config `babel-plugin-emotion` to use shorthand syntax with components, for now if I try  to use shorthand syntax like this (on www)

```
import { Link } from "gatsby"

const BlockLink = styled.Link`
  border-radius: ${presets.radiusLg}px;
  box-shadow: 0 1px 8px rgba(0, 0, 0, 0.2);
  display: flex;
  height: 100%;
  padding: ${rhythm(3 / 4)};

...
```
instead of `styled(Link)` I got the error, 
```
Uncaught Error: You're trying to use the styled shorthand without babel-plugin-emotion.
Please install and setup babel-plugin-emotion or use the function call syntax(`styled('Link')` instead of `styled.Link`)
```
so for consistency the full syntax is preferred,  we can think of shorthand syntax if it could be used with components not only with primitives

Reply:
👌Sticking to function call.